doc/source/whatsnew/v0.23.2.txt

```diff
@@ -59,6 +59,7 @@ Bug Fixes
 
 **I/O**
 
+- Bug in :func:`read_csv` when ``nrows=0``, ``low_memory=True``, and ``index_col`` was not ``None`` (:issue:`21141`)
```

Comment:
can you add a bit about that has changed from a user POV

Reply:
Sure thing.  Done.